### ISSUE TITLE
Please sign in again? - Way to keep non-idle? 

### ISSUE THREAD

**Issue by jmergy** [IsmaelMartinez/teams-for-linux#1357]
Assuming this is due to org policy around timeout. Can we do something that would keep the connection active enough to not trigger this? Native Teams or Outlook or other apps seem to be able to hold on to this. Maybe through certs or something? Pretty annoying. I understand there is only so much one can do here but asking.

**Comment by IsmaelMartinez** [comment]
Hi @jmergy , we had a similar issue in here https://github.com/IsmaelMartinez/teams-for-linux/issues/1045 but people reported it was better.

Maybe see if you can use the intune option.

**Comment by IsmaelMartinez** [comment]
did you manage to try and/or use one of the other options available to login? How did you get along with it? Thanks

**Comment by jmergy** [comment]
Still playing with it. We can close this for now and I will come back if I have some progress.

**Comment by masc-ppi** [comment]
I have the same problem, need to re-login every few hours. Our company uses Intune, and I tried the new settings for that. It does not make a difference. In the log, I get the following warning:
```
[2024-08-29 09:52:58.299] [debug] configFile: {
  ssoInTuneEnabled: true,
  ssoInTuneAuthUser: 'xxx@yyy.zzz'
}
[2024-08-29 09:52:58.786] [debug] Initializing InTune SSO
[2024-08-29 09:52:58.973] [debug] processArgs: [ '/opt/teams-for-linux/teams-for-linux' ]
[2024-08-29 09:52:58.974] [debug] testing RegExp processArgs false
[2024-08-29 09:52:58.975] [debug] Testing network using net.request() for https://teams.microsoft.com/v2
[2024-08-29 09:52:58.979] [debug] Spellchecker is disabled!
[2024-08-29 09:52:59.315] [warn]  Failed to find microsoft-identity-broker DBus interface
```
Do I need to start any other service/component to provide the missing microsoft-identity-broker DBus interface?

**Comment by IsmaelMartinez** [comment]
maybe check in here ? https://github.com/IsmaelMartinez/teams-for-linux/pull/1280 . There it explain how they did it and it might give you clues on how to get it to work.

**Comment by masc-ppi** [comment]
Thanks for pointing that out. I've checked #1280 and I'm now under the impression that I need to run the MS Intune app (https://learn.microsoft.com/de-de/mem/intune/user-help/microsoft-intune-app-linux). That one seems to provide the relevant DBus interface. Unfortunately, it is not yet available for Ubuntu 24.04.

**Comment by IsmaelMartinez** [comment]
I am going to close this as there is little more we can do to avoid this please sign in again other than the certificate and intune options already mentioned. Hopefully they release the intune app for ubuntu 24.04 soon.

**Comment by bpozdena** [comment]
@IsmaelMartinez  My problem is that I need to re-login via 3rd party SAML IdP with token every 24 hours, so there is no way to work around it. 

Questions:
1) Would it be possible to show a pop up message when the authentication expires? My problem is that I never know that I have been logged out when Teams is minimized. 
2) Would it be possible to automatically log out and ask for credentials when Teams start in the morning? That way I'm not silently deauthenticated at random times during the day? Alternatively, what do I need to delete every morning to get de-authenticated in Teams?
3) Is there any way to share the cookies from browser, where I'm already logged on via SAML and to Office365, etc?
4) Theoretically, if I create external script to authenticate me and extract the cookies, where and how I should inject them into Teams?

**Comment by martinsznapka** [comment]
I am experiencing same problem like bpozdena.

I turn on Teams4Linux in the morning and sometimes login dialog appears, sometimes not (probably depending on some token expiration).

If the login popup is not shown, then randomly after few hours I get suddenly toolbar with "Sign in" button, sometimes it drop me from meeting. It is very annoying. Then I need to login via 2F authentication with company credentials.

I would be happy if T4L force login on every startup so I do not experience this ugly behavior. I was searching for some parameters to do that, but no success.

**Comment by bpozdena** [comment]
After many missed messages during my working hours, I at least for now configured Teams to autostart maximized and with cleared storage using the `--minimized false --clearStorage` arguments. This way I don't forget to re-login every morning. 

Example:

cat ~/.config/autostart/teams-for-linux.desktop 

```
[Desktop Entry]
Categories=Chat;Network;Office;
Comment[en_US]=Unofficial Microsoft Teams client for Linux using Electron. It uses the Web App and wraps it as a standalone application using Electron.
Comment=Unofficial Microsoft Teams client for Linux using Electron. It uses the Web App and wraps it as a standalone application using Electron.
Exec=/opt/teams-for-linux/teams-for-linux --minimized false --clearStorage %U
GenericName[en_US]=
GenericName=
Icon=teams-for-linux
MimeType=
Name[en_US]=Teams for Linux
Name=Teams for Linux
Path=
StartupNotify=true
StartupWMClass=teams-for-linux
Terminal=false
TerminalOptions=
Type=Application
X-KDE-SubstituteUID=false
X-KDE-Username=

```

**Comment by bpozdena** [comment]
@IsmaelMartinez , would it be possible to implement OAuth2 authentication for custom "[MyApp](https://myapps.microsoft.com/)" similar to how [onedrive client for linux does it](https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#authorise-the-application-with-your-microsoft-onedrive-account)

With this approach my onedrive client remains logged in for weeks without any issues (until I am forced to change password).

**Comment by IsmaelMartinez** [comment]
not that easy as we use electron to wrap around this. I had a look to see if I could do something in the website part, but I can't seem to find a place where this would work. 

@martinsznapka , what happens if you close the app and clean the cache ? We might be able to force a cache cleanup (hopefully only of the authentication tokens part) but I am not sure where that authentication tokens sit. The browser section (I assume) but dunno where and if they are accessible

**Comment by martinsznapka** [comment]
I tried to Quit with Clear storage and opening T4L forced me to login. So clearing only authentication part would be great feature!

Clearing full storage isn't really nice for me, since some settings are gone (background image during meeting, predefined user login, etc.).

**Comment by IsmaelMartinez** [comment]
well, that gives me an area to explore ;)

**Comment by IsmaelMartinez** [comment]
Hi all,

I have added a debug log of the auth cookie changes in:
https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v1.11.2

It logs the cookie name, expiration date (in timestamp format) and domain.

Can you run the app with debug mode and see if we get more info from it? The cookie value is not printed at the moment, but can change that later if preferred. 

Trying to find a pattern in here as MS does support many ways of authentication so this cookie might not be what we are looking for.

**Comment by martinsznapka** [comment]
```
$ ./teams-for-linux --logConfig={"transports":{"console":{"level":"debug"},"file":{"level":false}}}
No config file found, using default values
all good with appLogLevels you aren't using them
all good with onlineCheckMethod you aren't using them
Initialising logger with config: ["{transports:console:{level:debug}}","{transports:file:{level:false}}"]
12:59:58.813 › File logging at /home/msznapka/.config/teams-for-linux/logs/main.log with level silly
12:59:58.814 › Console logging enabled with level silly
12:59:58.814 › Logger initialised
12:59:58.814 › configPath: /home/msznapka/.config/teams-for-linux
12:59:58.815 › configFile: {}
12:59:58.952 › processArgs: [
  '/home/msznapka/programs/teams/teams-for-linux',
  '--logConfig={transports:console:{level:debug}}',
  '--logConfig={transports:file:{level:false}}'
]
12:59:58.952 › testing RegExp processArgs false
12:59:58.953 › testing RegExp processArgs false
12:59:58.953 › testing RegExp processArgs false
12:59:58.953 › Waiting for network...
12:59:58.953 › Testing network using net.request() for https://teams.microsoft.com
12:59:58.955 › Spellchecker is disabled!
[210533:1016/125959.153733:ERROR:nss_util.cc(345)] After loading Root Certs, loaded==false: NSS error code: -8018
12:59:59.230 › Network test successful with method https
12:59:59.658 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 1s, IdleState: 'active'
[210533:1016/125959.660121:ERROR:object_proxy.cc(576)] Failed to call method: org.freedesktop.ScreenSaver.GetActive: object_path= /org/freedesktop/ScreenSaver: org.freedesktop.DBus.Error.NotSupported: This method is not implemented
13:00:00.105 › did-frame-finish-load {} true
13:00:00.106 › did-finish-load
13:00:00.843 › cookie changed cause: overwrite 
 removed?: true 

13:00:00.843 › cookie: authtoken 
 expirationDate: 1729135342.74778 
 domain: teams.microsoft.com
13:00:00.843 › cookie changed cause: explicit 
 removed?: false 

13:00:00.844 › cookie: authtoken 
 expirationDate: 1729135341.842887 
 domain: teams.microsoft.com
13:00:01.058 › Badge count set to '0'
13:00:01.065 › Badge count set to '0'
13:00:01.714 › Badge count set to '0'
13:00:03.390 › did-frame-finish-load {} false
13:00:09.719 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 1s, IdleState: 'active'
```

**Comment by IsmaelMartinez** [comment]
would you be able to leave the app running and see what logs we get on those cookies tomorrow? I think we might be able to detect the authentication token didn't refresh if we get a :
```
13:00:00.843 › cookie changed cause: overwrite 
 removed?: true 

13:00:00.843 › cookie: authtoken 
 expirationDate: 1729135342.74778 
 domain: teams.microsoft.com
````

That is not closely followed by a:
```
13:00:00.843 › cookie changed cause: explicit 
 removed?: false 

13:00:00.844 › cookie: authtoken 
 expirationDate: 1729135341.842887 
 domain: teams.microsoft.com
```

Then we can decide how to deal with it if that is a symptom

**Comment by martinsznapka** [comment]
```
$ ./teams-for-linux --logConfig={"transports":{"console":{"level":"debug"},"file":{"level":false}}}
No config file found, using default values
all good with appLogLevels you aren't using them
all good with onlineCheckMethod you aren't using them
Initialising logger with config: ["{transports:console:{level:debug}}","{transports:file:{level:false}}"]
08:54:39.215 › File logging at /home/msznapka/.config/teams-for-linux/logs/main.log with level silly
08:54:39.216 › Console logging enabled with level silly
08:54:39.216 › Logger initialised
08:54:39.217 › configPath: /home/msznapka/.config/teams-for-linux
08:54:39.217 › configFile: {}
08:54:39.351 › processArgs: [
  '/home/msznapka/programs/teams/teams-for-linux',
  '--logConfig={transports:console:{level:debug}}',
  '--logConfig={transports:file:{level:false}}'
]
08:54:39.352 › testing RegExp processArgs false
08:54:39.352 › testing RegExp processArgs false
08:54:39.352 › testing RegExp processArgs false
08:54:39.352 › Waiting for network...
08:54:39.352 › Testing network using net.request() for https://teams.microsoft.com
08:54:39.354 › Spellchecker is disabled!
[8443:1017/085439.535822:ERROR:nss_util.cc(345)] After loading Root Certs, loaded==false: NSS error code: -8018
08:54:39.595 › Network test successful with method https
08:54:39.965 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[8443:1017/085439.966405:ERROR:object_proxy.cc(576)] Failed to call method: org.freedesktop.ScreenSaver.GetActive: object_path= /org/freedesktop/ScreenSaver: org.freedesktop.DBus.Error.NotSupported: This method is not implemented
08:54:42.397 › did-frame-finish-load {} true
08:54:42.397 › did-finish-load
08:54:43.421 › did-frame-finish-load {} false
08:54:43.665 › Badge count set to '0'
08:54:43.674 › Badge count set to '0'
08:54:44.385 › Badge count set to '0'
08:54:44.600 › assignOnDidFailLoadEventHandler : {} - -3 - 
08:54:44.962 › did-frame-finish-load {} false
08:54:45.402 › assignOnDidFailLoadEventHandler : {} - -3 - 
08:54:45.803 › did-frame-finish-load {} false
08:54:46.019 › did-frame-finish-load {} false
08:54:46.256 › assignOnDidFailLoadEventHandler : {} - -3 - 
08:54:46.596 › did-frame-finish-load {} false
08:54:46.887 › assignOnDidFailLoadEventHandler : {} - -3 - 
08:54:48.972 › did-frame-finish-load {} false
08:54:49.219 › assignOnDidFailLoadEventHandler : {} - -3 - 
08:54:49.502 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
08:54:49.551 › did-frame-finish-load {} true
08:54:49.551 › did-finish-load
08:54:54.392 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
08:54:54.421 › did-frame-finish-load {} true
08:54:54.422 › did-finish-load
08:55:04.394 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 10s, IdleState: 'active'
08:55:04.521 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 10s, IdleState: 'active'
08:55:04.879 › did-frame-finish-load {} true
08:55:04.879 › did-finish-load
08:55:05.423 › did-frame-finish-load {} false
08:55:05.782 › Badge count set to '0'
08:55:05.793 › Badge count set to '0'
08:55:06.023 › assignOnDidFailLoadEventHandler : {} - -3 - 
08:55:06.474 › cookie changed cause: explicit 
 removed?: false 

08:55:06.474 › cookie: authtoken 
 expirationDate: 1729229047.474162 
 domain: teams.microsoft.com
08:55:06.474 › cookie changed cause: explicit 
 removed?: false 

08:55:06.474 › cookie: authtoken 
 expirationDate: 1729229047.474162 
 domain: teams.microsoft.com
08:55:06.475 › cookie changed cause: explicit 
 removed?: false 

08:55:06.475 › cookie: authtoken 
 expirationDate: 1729229047.474162 
 domain: teams.microsoft.com
08:55:06.475 › cookie changed cause: explicit 
 removed?: false 

08:55:06.475 › cookie: authtoken 
 expirationDate: 1729229047.474162 
 domain: teams.microsoft.com
08:55:06.535 › Badge count set to '0'
08:55:07.968 › Badge count set to '1'
08:55:08.207 › did-frame-finish-load {} false
08:55:08.382 › Badge count set to '3'
08:55:14.581 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
08:55:24.584 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 2s, IdleState: 'active'
08:55:29.375 › window close
08:55:29.509 › window closed
08:55:29.509 › before-quit
08:55:29.509 › will-quit
08:55:29.509 › quit
```

**Comment by IsmaelMartinez** [comment]
I wonder if this is on the back of locking your screen for an extended period. Aka, the re-authentication flow stops happening and when you unlock your screen the token has expired.

A few things to try and send logs. Start the app...
* And then close the laptop for a few hours. Reopen and share the logs 
* With other option for the lock screen inhibitor (screenLockInhibitionMethod). It defaults to Electron so try WakeLockSentinel that is the other option available. Similarly, share the logs.

Thanks!

**Comment by ciros88** [comment]
same issue here, keeps asking for signin every day

**Comment by IsmaelMartinez** [comment]
@ciros88 , can you provide the logs with the tests I indicated before? Still trying to figure out if we can detect this and force an refresh of the token. Otherwise you should focus on getting intune configured.

**Comment by GenocideStomper** [comment]
I have this same issue, the login stops working after pretty much exactly 24 hours and requires me to authenticate again. However in a browser such as Firefox I stay logged in for months.

For example, if I authenticate at 15:00, I can then turn off the PC for 12 hours, open the app the next day and still be logged in as long as it's before 15:00. At 15:00 it requires me to authenticate again.

I'm using the flathub version.

**Comment by GenocideStomper** [comment]
@IsmaelMartinez  I just /home/username to ~ in the log below.

```
$ flatpak run com.github.IsmaelMartinez.teams_for_linux --logConfig={"transports":{"console":{"level":"debug"},"file":{"level":false}}}[2:0109/144634.279951:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
No config file found, using default values
all good with appLogLevels you aren't using them
Initialising logger with config: ["{transports:console:{level:debug}}","{transports:file:{level:false}}"]
14:46:34.459 › File logging at ~/.var/app/com.github.IsmaelMartinez.teams_for_linux/config/teams-for-linux/logs/main.log with level silly
14:46:34.460 › Console logging enabled with level silly
14:46:34.461 › Logger initialised
14:46:34.461 › configPath: ~/.var/app/com.github.IsmaelMartinez.teams_for_linux/config/teams-for-linux
14:46:34.461 › configFile: {}
LaunchProcess: failed to execvp:
xdg-settings
Gtk-Message: 14:46:34.540: Failed to load module "xapp-gtk3-module"
Gtk-Message: 14:46:34.540: Failed to load module "canberra-gtk-module"
Gtk-Message: 14:46:34.662: Failed to load module "colorreload-gtk-module"
Gtk-Message: 14:46:34.662: Failed to load module "window-decorations-gtk-module"
[2:0109/144634.673007:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0109/144634.673060:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
14:46:34.736 › processArgs: [
  '/app/teams-for-linux/teams-for-linux',
  '--logConfig={transports:console:{level:debug}}',
  '--logConfig={transports:file:{level:false}}'
]
14:46:34.737 › testing RegExp processArgs false
14:46:34.737 › testing RegExp processArgs false
14:46:34.737 › testing RegExp processArgs false
[2:0109/144634.737690:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0109/144634.737721:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0109/144634.737743:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
14:46:34.737 › Waiting for network...
14:46:34.737 › Testing network using net.request() for https://teams.microsoft.com
14:46:34.739 › Spellchecker is disabled!
src/gbm_drv_common.c:131: GBM-DRV error (get_bytes_per_component): Unknown or not supported format: 808530000

src/gbm_drv_common.c:131: GBM-DRV error (get_bytes_per_component): Unknown or not supported format: 808530000

src/gbm_drv_common.c:131: GBM-DRV error (get_bytes_per_component): Unknown or not supported format: 808530000

src/gbm_drv_common.c:131: GBM-DRV error (get_bytes_per_component): Unknown or not supported format: 808530000

src/gbm_drv_common.c:131: GBM-DRV error (get_bytes_per_component): Unknown or not supported format: 808530000

src/gbm_drv_common.c:131: GBM-DRV error (get_bytes_per_component): Unknown or not supported format: 808530000

src/gbm_drv_common.c:131: GBM-DRV error (get_bytes_per_component): Unknown or not supported format: 808530000

src/gbm_drv_common.c:131: GBM-DRV error (get_bytes_per_component): Unknown or not supported format: 808530000

src/gbm_drv_common.c:131: GBM-DRV error (get_bytes_per_component): Unknown or not supported format: 808530000

src/gbm_drv_common.c:131: GBM-DRV error (get_bytes_per_component): Unknown or not supported format: 808530000

14:46:35.606 › Network test successful with method https
14:46:37.937 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
14:46:38.183 › did-frame-finish-load {} true
14:46:38.184 › did-finish-load
14:46:38.677 › did-frame-finish-load {} false
14:46:38.970 › Badge count set to '0'
14:46:38.991 › Badge count set to '0'
14:46:39.494 › Badge count set to '0'
14:46:39.555 › assignOnDidFailLoadEventHandler : {} - -3 - 
14:46:39.873 › did-frame-finish-load {} false
14:46:40.502 › assignOnDidFailLoadEventHandler : {} - -3 - 
14:46:40.822 › did-frame-finish-load {} false
14:46:40.937 › did-frame-finish-load {} false
14:46:41.156 › assignOnDidFailLoadEventHandler : {} - -3 - 
14:46:48.052 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
14:46:58.053 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
14:47:08.113 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
```

Afterwards I logged in again:
```
14:50:46.113 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
14:50:57.113 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
14:51:05.627 › did-frame-finish-load {} false
14:51:06.048 › assignOnDidFailLoadEventHandler : {} - -3 - 
14:51:06.325 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
14:51:06.475 › did-frame-finish-load {} true
14:51:06.475 › did-finish-load
14:51:11.864 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
14:51:11.922 › did-frame-finish-load {} true
14:51:11.922 › did-finish-load
14:51:16.032 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
14:51:16.224 › did-frame-finish-load {} true
14:51:16.225 › did-finish-load
14:51:16.709 › did-frame-finish-load {} false
14:51:16.948 › Badge count set to '0'
14:51:16.955 › Badge count set to '0'
14:51:17.196 › assignOnDidFailLoadEventHandler : {} - -3 - 
14:51:17.505 › Badge count set to '0'
14:51:18.711 › cookie changed cause: explicit 
 removed?: false 

14:51:18.711 › cookie: authtoken 
 expirationDate: 1736434277.711386 
 domain: teams.microsoft.com
14:51:18.711 › cookie changed cause: explicit 
 removed?: false 

14:51:18.711 › cookie: authtoken 
 expirationDate: 1736434277.711386 
 domain: teams.microsoft.com
14:51:18.712 › cookie changed cause: explicit 
 removed?: false 

14:51:18.712 › cookie: authtoken 
 expirationDate: 1736434277.711386 
 domain: teams.microsoft.com
14:51:18.712 › cookie changed cause: explicit 
 removed?: false 

14:51:18.712 › cookie: authtoken 
 expirationDate: 1736434277.711386 
 domain: teams.microsoft.com
14:51:18.882 › did-frame-finish-load {} false
14:51:19.055 › Badge count set to '2'
14:51:23.325 › Badge count set to '1'
14:51:24.331 › did-frame-finish-load {} false
14:51:26.094 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
14:51:26.175 › did-frame-finish-load {} false
14:51:36.095 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'

```

**Comment by ciros88** [comment]
sorry for the delay @IsmaelMartinez, here's my logs on Manjaro (with the "We need you to sign in again. This could be a request from your IT department or Teams, or the result of a password update." appearing message):

```
$ teams-for-linux --logConfig={"transports":{"console":{"level":"debug"},"file":{"level":false}}}
all good with appLogLevels you aren't using them
Initialising logger with config: ["{transports:console:{level:debug}}","{transports:file:{level:false}}"]
11:03:34.275 › File logging at /home/ciros/.config/teams-for-linux/logs/main.log with level silly
11:03:34.276 › Console logging enabled with level silly
11:03:34.277 › Logger initialised
11:03:34.277 › configPath: /home/ciros/.config/teams-for-linux
11:03:34.277 › configFile: { disableNotificationSound: true }
11:03:34.777 › processArgs: [
  '/opt/teams-for-linux/teams-for-linux',
  '--logConfig={transports:console:{level:debug}}',
  '--logConfig={transports:file:{level:false}}'
]
11:03:34.778 › testing RegExp processArgs false
11:03:34.778 › testing RegExp processArgs false
11:03:34.778 › testing RegExp processArgs false
11:03:34.779 › Waiting for network...
11:03:34.779 › Testing network using net.request() for https://teams.microsoft.com
11:03:34.782 › Spellchecker is disabled!
11:03:35.293 › Network test successful with method https
11:03:35.688 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
11:03:35.989 › did-frame-finish-load {} true
11:03:35.989 › did-finish-load
11:03:37.248 › Badge count set to '0'
11:03:37.263 › Badge count set to '0'
11:03:38.241 › Badge count set to '0'
11:03:39.705 › did-frame-finish-load {} false
11:03:45.760 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
```



after the signin:
```
11:08:09.324 › did-frame-finish-load {} false
11:08:09.566 › assignOnDidFailLoadEventHandler : {} - -3 -
11:08:09.798 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
11:08:09.866 › did-frame-finish-load {} true
11:08:09.867 › did-finish-load
11:08:18.034 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
11:08:18.300 › did-frame-finish-load {} true
11:08:18.301 › did-finish-load
11:08:19.618 › Badge count set to '0'
11:08:19.629 › Badge count set to '0'
11:08:20.608 › Badge count set to '0'
11:08:20.665 › cookie changed cause: explicit
 removed?: false

11:08:20.665 › cookie: authtoken
 expirationDate: 1737281117.664891
 domain: teams.microsoft.com
11:08:20.665 › cookie changed cause: explicit
 removed?: false

11:08:20.665 › cookie: authtoken
 expirationDate: 1737281117.664891
 domain: teams.microsoft.com
11:08:20.666 › cookie changed cause: explicit
 removed?: false

11:08:20.666 › cookie: authtoken
 expirationDate: 1737281117.664891
 domain: teams.microsoft.com
11:08:22.243 › did-frame-finish-load {} false
11:08:28.105 › GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
```

**Comment by IsmaelMartinez** [comment]
Can you check if the following release improves the situation? 

[1.12.7](https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v1.12.7). 

I have been running it for over 24 hours and the session seems to work fine (refreshing).

But it might just be a side effect on me having the MacOS app also running.

**Comment by jmergy** [comment]
Oh cool - I will check it out. I have TFL running in separate instances with some rather aggressive time-outs so I will see.

**Comment by IsmaelMartinez** [comment]
If it doesn't work, it would be great if you can just open the webdebug and then run the following:

```
await document.getElementById('app')._reactRootContainer.current.updateQueue.baseState.element.props.coreServices.authenticationService._coreAuthService._authProvider.acquireToken("https://graph.microsoft.com", { correlation: document.getElementById('app')._reactRootContainer.current.updateQueue.baseState.element.props.coreServices.correlation, forceRenew: true} )
```

Do NOT share that token but see if that then makes your session last 'double' the usual time. If that is the case, then we can add some logic to call that function every now an then.

I hope to play with that token to get some notifications from here https://learn.microsoft.com/en-us/graph/teams-change-notification-in-microsoft-teams-overview , but I don't know when I will be able to do this.

**Comment by GenocideStomper** [comment]
> Can you check if the following release improves the situation?
> 
> [1.12.7](https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v1.12.7).


Is this getting pushed to flathub soon, or any way I could install the pre-release version through flatpak? Would love to try it in the same conditions.

**Comment by GenocideStomper** [comment]
Completely understood! I will try the appimage version soon, may take a few days for my feedback.

**Comment by GenocideStomper** [comment]
Didn't work, still asked to login again.

I'm running the AppImage on Plasma 6 wayland and X11, and I'm not getting a taskbar icon so I don't know how open the webdebug

**Comment by martinsznapka** [comment]
22.01 9AM - installed 1.12.7
22.01 9AM - Sign In requested
23.01 9AM - Sign In requested
24.01 9AM - Sign In requested

**Comment by IsmaelMartinez** [comment]
Yeah, that didn't work.

It is a pain it takes a day to see the results. Do any of you know a way to make it happen faster?   

Thanks

**Comment by GenocideStomper** [comment]
Unfortunately not. I guess the only thing would be to test in parallel: different app types, on different PCs / VMs

But thank you for having a look at this issue!

**Comment by IsmaelMartinez** [comment]
I will keep digging. Can I ask what URL you see when you go to the website? does it finish with /v2? Thanks

**Comment by IsmaelMartinez** [comment]
@GenocideStomper, can you check if you got options --url (with https://teams.microsoft.com/v2) and --optInTeamsV2 

See https://github.com/IsmaelMartinez/teams-for-linux/blob/main/app/config/README.md for more details. 

I wonder if is getting confused as without that options it will try to go to teams-v1 and then get redirected. It should not matter, but something else to test.

**Comment by GenocideStomper** [comment]
Sorry, this is embarrassing after advocating looking at Firefox, but suddenly it's asking me to authenticate daily there as well.

Do you still want me to test what you wrote in your last comment?

**Comment by IsmaelMartinez** [comment]
If you can try it, that would be great. My thinking is that the token they generate is not valid for renewal for teams.microsoft.com/v2 but for teams.microsoft.com . The reauthentication flow then is not triggering. 

I suspect is a microsoft issue/change as I do get the exact same problem in the MacOS app. Is pretty annoying tbh

**Comment by GenocideStomper** [comment]
Neither 
`flatpak run com.github.IsmaelMartinez.teams_for_linux --optInTeamsV2=false --url='https://teams.microsoft.com/'`
nor
`flatpak run com.github.IsmaelMartinez.teams_for_linux --optInTeamsV2=true --url='https://teams.microsoft.com/v2'`
made any difference. I think I always get sent to the same version, which I think is v2.

**Comment by martinsznapka** [comment]
I am running these two params for about 6 days and no difference. Still same 24h cycle of token expiration.

**Comment by IsmaelMartinez** [comment]
Ok, had a bigger dig around and I suspect is something to do with 3rd party cookies. 

I can see others having the same issue:

https://github.com/electron/electron/issues/19825
https://github.com/ferdium/ferdium-app/issues/522
https://github.com/electron/electron/issues/36922
https://github.com/ferdium/ferdium-app/issues/1171

And seems to be pointing here https://learn.microsoft.com/en-us/microsoftteams/troubleshoot/teams-sign-in/sign-in-loop#resolution

I randomly came round to that after reading in https://learn.microsoft.com/en-us/entra/identity-platform/reference-third-party-cookies-spas but I got something testing, but you know this takes a while. I let you know how it goes

**Comment by IsmaelMartinez** [comment]
Added a few extra checks in 1.12.8.

I am not sure if it will fix the situation, but do have a try. Sorry but it is just an annoying issue and I haven't been able to do much in the last few weeks. Life sometimes gets busier than I would like.

**Comment by GenocideStomper** [comment]
Sorry for the delay, I finally got around to testing 1.12.8 and still have the same issue 😕

**Comment by Flofler** [comment]
@IsmaelMartinez Can I suggest another change? It doesn't mind me at all to relogin every morning, so I activated the clearStorage option but it's very annoying, that I have to reconfigure everything else (like video background, new Teams calender option, ...). Is it possible to implement an option to only delete the session cookie so I just have to relogin without clearing everything else?

**Comment by IsmaelMartinez** [comment]
Hi all, 

Having a wee look I think this is a MS issue, rather than the app. https://answers.microsoft.com/en-us/msteams/forum/all/banner-in-new-teams-desktop-please-sign-in-again/e183eceb-01c1-4fdc-afdb-c39aee54405e?page=1 and https://feedbackportal.microsoft.com/feedback/idea/d1c8d972-8f55-ef11-b4ad-000d3a0e6a3a

I have added a few more permissions allowed and domains, but I think I should remove that and the cookies logging as it only adds noise to the logs.

I will check if I can do your suggestion @Flofler , and if this can be automated. I suspect you all get this a log message like 
```
› assignOnDidFailLoadEventHandler : {} - -3 - 
```

If that is the case, I can see if we can "act" on and redirect you all to the login page. I think that is related to the authentication not been there, but it is a bit confusing as some of you don't show that message on the logs, but do see the banner.

Can some of you check the [1.13.2](https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v1.13.2) and see if it improves the situation? (I don't have much hopes on it). Thanks!

**Comment by martinsznapka** [comment]
I used 1.13.2 and got "Sign in" panel in a standard expected time (24h after last login).
So no change.

**Comment by IsmaelMartinez** [comment]
I do think is that Microsoft issue. Reason no matter what we do, it is still broken. Hopefully they get their act together and fix it and/or we can make it a bit more smooth for you guys.

**Comment by martinsznapka** [comment]
I think the proposal from Flofler is pretty nice and would work for me too: https://github.com/IsmaelMartinez/teams-for-linux/issues/1357#issuecomment-2735748679

**Comment by IsmaelMartinez** [comment]
I have added that as an option. See #1605 for reference and you can play with it in  https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v1.14.2 Hope that helps a wee bit!

**Comment by martinsznapka** [comment]
Hi, thx for new version. I created new file `~/.config/teams-for-linux/config.json` with content
```
{
  "clearStorageData": {
    "storages": [
      "cookies",
      "sessionStorage"
    ],
    "quotas": [
      "temporary"
    ]
  }
}
```
But starting the app will not ask me for credentials, simply it starts already signed-in.
Any idea?

**Comment by IsmaelMartinez** [comment]
Strange, do you mind opening a new ticket with this and the logs? Just to not muddle the info in this ticket and the other. Maybe you are using flatpak or others and the config file is in a different location.

**Comment by Flofler** [comment]
@martinsznapka it does work for me with the following configuration:
```
"clearStorageData": {
        "storages": ["localstorage"],
        "quotas": ["temporary"]
}
```

As far as I can see `sessionStorage` is no valid type of storage in Electron.

> The types of storages to clear, can be cookies, filesystem, indexdb, localstorage, shadercache, websql, serviceworkers, cachestorage

**Comment by jmergy** [comment]
still would love a way to trick this, but ...

**Comment by joakim-tjernlund** [comment]
We also need to resign in everyday and it is getting really boring to do that.
One thing that would help if the password could be remembered as long as you don't restart Teams.

Using browsers, the password is remembered so maybe Teams-for-linux can too?

**Comment by IsmaelMartinez** [comment]
Hi @joakim-tjernlund , unfortunately that doesn't seem to be the case. I don't really know why this is happening tbh, but maybe microsoft doesn't like the user agent we use and don't refresh the token (or that is what it feels like). We do use a persistent session, what should mean your session is still available even if you close the session (it should keep the cookies and the likes).

**Comment by IsmaelMartinez** [comment]
we don't do anything in particular to not remember passwords etc. Our current implementation should be doing this, but there is something in microsoft side that is making this not work anymore.

**Comment by Flofler** [comment]
@joakim-tjernlund I can really recommend the autotype feature of KeepassXC. It's much easier to relogin every day.

**Comment by joakim-tjernlund** [comment]
> [@joakim-tjernlund](https://github.com/joakim-tjernlund) I can really recommend the autotype feature of KeepassXC. It's much easier to relogin every day.

That would work for me but passing that around to all my users at work would be hard work, yet another tool to maintain too.

**Comment by IsmaelMartinez** [comment]
As far as I am aware, what they do is store a session cookie or token that gets refreshed regularly.

**Comment by IsmaelMartinez** [comment]
To give more context. We will need to intercept the password, check the page every second or so to see if there is a password field in it, then fill it with the plain text password. 

as Microsoft changes those fields on a regular basis, and they can also be different between companies (depending on the login flow they decide) this will be flakey and difficult to test. 

It can then easily become an attack vector (someone making you open a crafted page that has those same fields). 

Over all, I don't like going down that route, but thanks for bringing the suggestion. The token refresh should work but maybe teams-for-linux is not deemed as "secure" by your it admin and is not refreshing the token, or maybe is something else.

**Comment by martinsznapka** [comment]
> To give more context. We will need to intercept the password, check the page every second or so to see if there is a password field in it, then fill it with the plain text password.
> 
> as Microsoft changes those fields on a regular basis, and they can also be different between companies (depending on the login flow they decide) this will be flakey and difficult to test.
> 
> It can then easily become an attack vector (someone making you open a crafted page that has those same fields).
> 
> Over all, I don't like going down that route, but thanks for bringing the suggestion. The token refresh should work but maybe teams-for-linux is not deemed as "secure" by your it admin and is not refreshing the token, or maybe is something else.

Make sense. It would be overengineering and new security flaws.

I am quite happy with current situation: each time I start teams it asks me to login. The only disadvantage is I need to select video background after each login. My settings:
```
{
  "clearStorageData": {
    "storages": [
      "localstorage"
    ],
    "quotas": [
      "temporary"
    ]
  }
}
```
Anyone has suggestion how to keep background settings?

**Comment by pazepaze** [comment]
Together with our admin we found what was causing the session expiry every 24h with the red banner appearing at the top in teams-for-linux for me.

We observed: When logging in, I did not have to type my password (just 2FA was required) and the "keep me signed in" dialogue did not show up. This meant my session expired in 24h.

The cause of this behavior was the following: We are using a Microsoft feature called (Entra?) Seamless SSO. This allows some kind of easier login procedure for a known device from inside the company network. But for some reason this leads to the "keep me signed in" dialogue to be missing when I use my known device on the company network. I also have intune installed, not sure if this is involved here.

The workaround for now is, that I sign in to teams from outside the company network one time. This leads to the password check and "keep me signed in" dialogue/checkbox showing up during login, which gives me a session that, as my admin tells me, can stay active for 90 days instead of 24h.
I also have set ssoInTuneEnabled set to "false" during this process and am keeping it off now.

Maybe this will help others who work in a similar setup.

**Comment by IsmaelMartinez** [comment]
If it is of any consolation, I have the exact same issue with the official client in MacOS. I don't know if they change something, but been suffering this for the last few months (not as long as this ticket has been open, but from last year)

**Comment by GenocideStomper** [comment]
Comment to remove the stale label

**Comment by martinsznapka** [comment]
I tried the following workaround but it didn't help in my case. I logged in outside of VPN, but next day I saw again toolbar with Sign In.

Anyway, thx for sharing your story.

https://github.com/IsmaelMartinez/teams-for-linux/issues/1357#issuecomment-2875384595
> 
> The workaround for now is, that I sign in to teams from outside the company network one time. This leads to the password check and "keep me signed in" dialogue/checkbox showing up during login, which gives me a session that, as my admin tells me, can stay active for 90 days instead of 24h. I also have ssoInTuneEnabled set to "false" during this process and am keeping it off now.
> 
> Maybe this will help others who work in a similar setup.

**Comment by IsmaelMartinez** [comment]
As I mentioned, I have the exact same issue with the official MS teams what makes me think is more of a policy issue/configuration that the app has little to do. Hopefully we can find a workaround.

**Comment by MattHeffNT** [comment]
Don't ask me how/why it worked. But I have this installed via flatpak on PopOS. I uninstalled then reinstalled and it seems to have fixed the problem for me.  (I use teams for my University).

**Comment by IsmaelMartinez** [comment]
looks like there is another issues that relates to this. I will see when I can have a look but putting here for reference (and for this one not to be closed in 5 days) https://github.com/IsmaelMartinez/teams-for-linux/issues/1756

**Comment by fmoessbauer** [comment]
Hi, `linux-entra-sso` maintainer here. Together with MS support, we had an extensive analysis at the same issue in the teams browser app on Linux (e.g. in Firefox / Chrome). In short, the teams app uses the MSAL, which performs the background SSO in an `iframe`. This happens on expiry of the `refresh_token`, which is (depending on backend config) ~24h.

<details>
<summary>Reason why it was not working in linux-entra-sso</summary>

In case of the `linux-entra-sso` webext, this did not work, as the permission to inject data into requests was only given for `login.microsoftonline.com` but not for `teams.microsoft.com`, which is also needed in case of the iframe approach (fixed in https://github.com/siemens/linux-entra-sso/pull/75).
</details>

Now we have two situations (of course depending on the backend config):

**without Intune:** The user simply cannot silently request a new `refresh_token`, as he needs to reauthenticate - which is usually after ~24 hours.

**with Intune:** The background request is successful as long as the token is correctly injected, because the login can be performed without user interaction. I confirmed (with the dev-console), that the `teams-for-linux` app correctly injects the token into the `sub_frame`. By that, no manual re-auth is required, as long as the token can be acquired from the broker at time of re-auth. If that token acquisition fails for whatever reason, a manual re-auth is required.

While debugging this and inspecting the logs, I further noticed that the `teams-for-linux` app also injects the tokens into the XHR requests towards `login.microsoftonline.com` - which is not needed but creates a lot of traffic on the broker. I'm also wondering if this is silently ignored by the entra backend, or if this rings some bells. Once the we have enough evidence that the approach `linux-entra-sso` takes is correct, I'll also propose a patch regarding this here as well.

**PS:** The `teams-for-linux` could simply check for the expiry of the `refresh_token` in the local storage. If that is expired by some couple of seconds, we know that a manual re-login is required. In this case, we could simply restart the app.

**Comment by IsmaelMartinez** [comment]
Hi @fmoessbauer ,thanks for your message. I would see if the cache issue does somehow fix this, as we don't do anything special to handle authentication (nor I think we should need to).

I hope to be able to get something out at some point next week. Thanks everyone for contributing and hopefully we get this fixed soon one way or another

**Comment by IsmaelMartinez** [comment]
Can you check with version https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v2.1.2 ?

I have added a cache cleanup that you can enable with the following config option:

```
{
  "cacheManagement": {
    "enabled": true,
    "maxCacheSizeMB": 300,
    "cacheCheckIntervalMs": 3600000
  }
}
```

If this sorts the issue, I will make it the default, but want to make sure it doesn't break everyones app :D 

Fixed in https://github.com/IsmaelMartinez/teams-for-linux/issues/1756 

Thanks you all in advance and hopefully this solves the problem (or at least improves it)

**Comment by joakim-tjernlund** [comment]
Testing it now, any way to make this config system wide(so all users get managed cache)?

**Comment by IsmaelMartinez** [comment]
If it works and doesn't have undesired consequences this will be the default. (so all users will get it). Thanks for checking and hopefully in 24 hours you still have a session!

**Comment by jmergy** [comment]
testing as well.

**Comment by joakim-tjernlund** [comment]
> If it works and doesn't have undesired consequences this will be the default. (so all users will get it). Thanks for checking and hopefully in 24 hours you still have a session!

Great! But I still want to know how to do it globbaly. I manage our Linux at work and sometimes I need to change stuff globally for all.

BTW, here is what was printed when started
```
teams-for-linux 
Initialising logger with config: {"transports":{"console":{"level":"info"},"file":{"level":false}}}
20:22:28.469 › configPath: /home/jocke/.config/teams-for-linux
20:22:28.505 › (node:26858) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.
(Use `teams-for-linux --trace-deprecation ...` to show where the warning was created)
20:22:28.615 › 🧹 Cache Manager started { maxSize: '300MB', checkInterval: '3600s' }
20:22:31.215 › Cache size exceeded limit { currentSize: '1916MB', maxSize: '300MB' }
20:22:34.298 › assignOnDidFailLoadEventHandler : {} - -3 - 
20:22:36.375 › Cache cleanup completed { oldSize: '1916MB', newSize: '54MB', freed: '1862MB' }
20:22:36.708 › assignOnDidFailLoadEventHandler : {} - -3 - 
20:23:01.208 › assignOnDidFailLoadEventHandler : {} - -3 - 
20:24:14.109 › assignOnDidFailLoadEventHandler : {} - -3 - 
20:24:28.580 › assignOnDidFailLoadEventHandler : {} - -3 - 
```

**Comment by IsmaelMartinez** [comment]
we don't have a way to configure the config globaly, and I am not sure how we would be able to do that and making it distributable, but open a feature request and I can have a look. Put as much thought as you can, so I can see if it can be done... but it might be possible. The how to distribute that change is a bit more tricky, as you will need to somehow configure all instances at least once. 

Open that issue, again with as much details as you can, and we can discuss it. It should be possible... just need to find the right way

**Comment by IsmaelMartinez** [comment]
And I do hope @jmergy this one is also sorted for you. That cache cleanup log @joakim-tjernlund is what I was expecting and you had a lot of stuff in the cache. Do let me know if things are odd... the cache should clear hourly by default, but is configurable.

**Comment by Flofler** [comment]
> Testing it now, any way to make this config system wide(so all users get managed cache)?

I would recommend using a symlink to a shared directory that is used by all users in your company.

**Comment by IsmaelMartinez** [comment]
Simple is best, and that sounds like a simple solution.

**Comment by joakim-tjernlund** [comment]
yes, I was just thinking a config.json under say /etc/chromium/ which is read before users config.json.
I will handle distribution of said file.

**Comment by joakim-tjernlund** [comment]
> > Testing it now, any way to make this config system wide(so all users get managed cache)?
> 
> I would recommend using a symlink to a shared directory that is used by all users in your company.

where do I put this symlink? It cannot be under /home/<users>

**Comment by Flofler** [comment]
@joakim-tjernlund why not? I was thinking about replacing the configuration file `~/.config/teams-for-linux/config.json` with a symlink to a file under your control.

**Comment by joakim-tjernlund** [comment]
> [@joakim-tjernlund](https://github.com/joakim-tjernlund) why not? I was thinking about replacing the configuration file `~/.config/teams-for-linux/config.json` with a symlink to a file under your control.

Then I would have to run script that is triggered each time a new user logs in, messy.
On to of that I am destroying any config existing users already have made for themself.

**Comment by IsmaelMartinez** [comment]
by the way, unfortunately the sign issue seems to be still present. @joakim-tjernlund , I recommend you open another issue or discussion for that feature request, as otherwise issues get a bit messy with multiple issues/features.

**Comment by joakim-tjernlund** [comment]
>  [@joakim-tjernlund](https://github.com/joakim-tjernlund) , I recommend you open another issue or discussion for that feature request, as otherwise issues get a bit messy with multiple issues/features.

https://github.com/IsmaelMartinez/teams-for-linux/issues/1773

**Comment by IsmaelMartinez** [comment]
I have little faith, but this might be fixed in [2.5.0](https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v2.5.0). The reason is that the cache manager, that was meant to fix the issue if it was a cache overflow, was cleaning the parts of the app where the refresh token apparently gets stored.

If this doesn't work, I got some plans to add more logging and see if we can get it fixed soon. Apologies it taking so long, it is one of those issues that is difficult to understand and debug.

**Comment by FingerlessGlov3s** [comment]
> I have little faith, but this might be fixed in [2.5.0](https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v2.5.0). The reason is that the cache manager, that was meant to fix the issue if it was a cache overflow, was cleaning the parts of the app where the refresh token apparently gets stored.
> 
> If this doesn't work, I got some plans to add more logging and see if we can get it fixed soon. Apologies it taking so long, it is one of those issues that is difficult to understand and debug.

I've noticed when I do `Sign In` when I get a banner along the top of the teams application, I can sign and it's happy again. I did notice it doesn't pass the Device SSO/Identity through with that request, so it doesn't know I'm on a compliant Linux Device for evaluating Conditional Access policies.

I have also checked the `intune-agent.service` it's happy and communicating with Intune.

Am I supposed to see that behaviour of not seeing the device details come through?

<img width="690" height="250" alt="Image" src="https://github.com/user-attachments/assets/e111403e-c77e-4345-9e00-d8f0216ddab1" />

Is this also something maybe fixed in 2.5.0?

**Comment by IsmaelMartinez** [comment]
Hi @FingerlessGlov3s , do you have intune configured? If you use intune it should not ask you for password (I think). Check https://github.com/IsmaelMartinez/teams-for-linux/pulls?q=is%3Apr+intune+is%3Aclosed (not yet documented how to setup but is a couple of config options). If you have problems with that configuration, do open another ticket to keep this one on the re-sign in again issue focus only. Thanks!

**Comment by IsmaelMartinez** [comment]
Can you try with https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v2.5.3? If it doesn't work can you provide some logs?

I added more logging and an attempt to fix it. If it doesn't fix it the extra logging might help me identify the issue(hopefully). 

Same command:

```
ELECTRON_ENABLE_LOGGING=true teams-for-linux-electron --logConfig='{}' 
```

Thanks!

ps: apologies for the auto-close issue automation

**Comment by cpfeiffer** [comment]
Today, I didn't have to sign in with 2.5.3 🎉

**Comment by jmergy** [comment]
Running 2.5.3 but wondering if I have missed something. Is there anything special I should have in my config.json now to leverage something or? Right now, I am standard. I did have some cache settings in there a while back to try various but have removed all that. 

If there is something I need to tweak in config.json, let me know.

**Comment by IsmaelMartinez** [comment]
you shouldn't need to add anything to the config. If it isn't working for you, see if you can share the logs. @cpfeiffer , is this still working for you with 2.5.3? Ta

**Comment by IsmaelMartinez** [comment]
Hi, I have added more logs in https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v2.5.5 . Can some of you try it and send the logs? Hopefully this will give me more information that hopefully either confirm if this is fixed or how to fix it. 

Apologies for the noise on the automation (that I have disable it because is pretty annoying). Thanks in advance!

**Comment by IsmaelMartinez** [comment]
Hi @cpfeiffer and @jmergy , did you manage to test this for a few days? the current implementation on this version is 'fine' but I would like to tidy up the edges and improve it, but only if this is actually working. Thanks in advance!

**Comment by jmergy** [comment]
Thanks @IsmaelMartinez - I always have a couple of instances of teams-for-linux running to handle a couple of different user accounts. One instance with multiple connected/guest orgs is working fine. The other instance, a single org, with some very strict time-outs and an external MFA outside of Azure, using Duo, has been always challenging here. It could be that even in these developmental improvements, there could be something with it behind the scenes and with the caveat of the external auth MFA off-platform, this could just not be able to be overcome. 

I will keep using but thinking at this point it might not be able to be overcome due to these additional constraints. Please move ahead on any additional fixes and I will keep you posted.

**Comment by IsmaelMartinez** [comment]
I will then add the extra bits I wanted to add, and then consider this issue as completed (if it is actually improving the situation at least in some cases). 

I plan to explore querying the microsoft graph API and see if that opens options and/or simplifies a few areas from the app. Thanks for the quick reply (and for using the app)!

**Comment by IsmaelMartinez** [comment]
Ok, I am going to try to change the default refresh method, to force a refresh every hour. I don't see it happening as often as the refresh period (every hour), and there is some sort of threshold microsoft uses internally to determine if to refresh the token. Probably only refreshes the token if there are x hours left of it.

That means that if you are using the app continuously, the refresh seems to happens (maybe after 8/12 hours), but if you use it during working hours, then you will need to login again in the morning as you missed your refresh window. 

I will do 2 releases, one with a more secure way to store tokens (I think we can do better than local storage as I think the browser does) and then I will force a refresh every hours (and make it configurable). I let you know so you can test. Hopefully this then get sorted. Tricky one but I think we are getting there.

**Comment by FingerlessGlov3s** [comment]
> Thanks [@IsmaelMartinez](https://github.com/IsmaelMartinez) - I always have a couple of instances of teams-for-linux running to handle a couple of different user accounts. One instance with multiple connected/guest orgs is working fine. The other instance, a single org, with some very strict time-outs and an external MFA outside of Azure, using Duo, has been always challenging here. It could be that even in these developmental improvements, there could be something with it behind the scenes and with the caveat of the external auth MFA off-platform, this could just not be able to be overcome.
> 
> I will keep using but thinking at this point it might not be able to be overcome due to these additional constraints. Please move ahead on any additional fixes and I will keep you posted.

The instance that's a single org, do you know if they're applying session lengths limits in Conditional Access, if they are you'll be forced to MFA at the end of each period.  I know some tenants do this when you login from what is dubbed BYOD, being using your own personal device. Ideally you'd want to set the session period to be 8-12 hours or so to ensure people don't say signed in on non-corporate devices longer than needed, since you don't know if their device is shared, passworded, encrypted etc.

I use the intuneSSOAuth feature which passes through the device identity with the authentication requests, so when I look in Entra Sign-Ins I can see the exact device I'm signing in from and shows me using a corporate device, which I am. So then the extra BYOD policies then don't kick in. I stayed signed in up to 14+ hours, 9am to 11pmish. I always shutdown my work laptop.

**Comment by IsmaelMartinez** [comment]
hello, 

I have released [2.5.9](https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v2.5.9) that basically stores those tokens now in secure storage. That is not really necessary, as in the browser they are stored in the plain storage, but wanted to go a bit further with it.

This will probably result on your session expiring, but do let me know if the session has been active between days. I do have another branch with some changes to force the refresh of tokens, but if the current solution works, I might avoid adding that extra complexity (as I don't think it will solve that MFA one)

Thanks all for your help

**Comment by jmergy** [comment]
Testing 2.5.9 now. Thanks @IsmaelMartinez

**Comment by Flofler** [comment]
Unfortunately the problem still persists with version 2.5.9

**Comment by IsmaelMartinez** [comment]
Hi @Flofler, thanks for the feedback and sorry it is not working for you. Are you able to put some logs? Getting confuse as some people say it works while others don't. Are you able to try this branch? https://github.com/IsmaelMartinez/teams-for-linux/tree/feature/configurable-token-refresh 

If that works I can put it as an optional if it does work (I will need to do some work on it, but it should be functional and refresh all tokens every 15 minutes). The refresh of course can only happen if you computer is active with MS teams open. If it goes to sleep or you close it, the refresh stops. 

It might be better to start a new issue that references this one, as it has far too many comments now to keep track of the context and it has morph a wee bit.

**Comment by Flofler** [comment]
Unfortunately I didn't enabled logging. So I have no logs to provide to you. There is some improvement for me I think. It now consistently is logging me out every morning I've started my computer. Before it was at random times (maybe 24 hours after my last login). I can live with that.

Unfortunately I cannot test your branch at the moment as I'm not in office until next year. So I'm not really able to test if this works until then.

**Comment by joakim-tjernlund** [comment]
> Unfortunately I didn't enabled logging. So I have no logs to provide to you. There is some improvement for me I think. It now consistently is logging me out every morning I've started my computer. Before it was at random times (maybe 24 hours after my last login). I can live with that.

That could be your work policy mandating that.

**Comment by joakim-tjernlund** [comment]
With latest version I still see "sign in again", I have an idea why.
Every now and then there is some glicth or TMO happening and sign in again is needed(I never have to enter  OTP here).
Since Teams-for-linux does not cache password, it has to ask for it.
Browsers like Firefox has PW saved and will sign in for you automatically.

**Comment by IsmaelMartinez** [comment]
did you try that branch I put? That one will force a token refresh every 15 minutes. We do store the refresh tokens now as the browser does, but if you close your laptop/computer for an extended period the refresh doesn't happen and that triggers it to expire. Not sure how many hours (I hope it was 24 but I think it might be lower)

**Comment by joakim-tjernlund** [comment]
Just used latest release and no laptop, this is a desktop always on

**Comment by jmergy** [comment]
trying 2.5.13 now - the one instance with the very harse time-out policy is annoying but the overall value and daily constant use of teams-for-linux has been tremendous for me. Will be great if this works, but by no means a major issue. It would be cool to nix this though and appreciate all collective effort.

**Comment by IsmaelMartinez** [comment]
@joakim-tjernlund are you able to test that branch I mentioned? Not a problem if you can't. Thanks

**Comment by joakim-tjernlund** [comment]
> [@joakim-tjernlund](https://github.com/joakim-tjernlund) are you able to test that branch I mentioned? Not a problem if you can't. Thanks

That seems to be  just source? I cannot build Teams-for-linux

**Comment by IsmaelMartinez** [comment]
no problems, I don't think it does help anyway, but in case the forcing refresh do change something. If you can provide some logs of when the session fails to refrehs taht will be great. Thanks

**Comment by IsmaelMartinez** [comment]
I think is time to close this issue as there are more than 100 comments and there has been some fixes that has improved the issue at least for the reporter @jmergy . For the hard policies I don't think we can't do much. Appreciate if someone can check that branch and see if it improves it. If it does we can open another enhacement to provide that as an option . Appreciate all your help and hopefully you understand why I close the issue. It has been open for that long and has that much evolving context that is now confusing and contradictory. Thanks again for all the collective effort!

**Comment by jmergy** [comment]
Agree and thank you.
